src/trans/gpu/algor/growing_allocator_mod.F90

```diff
@@ -89,4 +83,20 @@ SUBROUTINE REGISTER_FREE_C(ALLOC_C, FREE_FUNC_C) BIND(C, NAME="growing_allocator
 
   END SUBROUTINE
 
+  SUBROUTINE DESTROY_GROWING_ALLOCATOR(ALLOC)
+    USE ISO_C_BINDING, ONLY: C_SIZE_T
+    IMPLICIT NONE
+    TYPE(GROWING_ALLOCATION_TYPE) :: ALLOC
+    INTEGER :: I
+    IF (ALLOCATED(ALLOC%PTR)) THEN
```

Comment:
I don't think this is strictly compliant - only `ALLOCATABLE` objects can be passed to `ALLOCATED`. Somehow nvfortran doesn't mind, but this line fails when compiling with Cray.

You can of course check whether a pointer is `ASSOCIATED`. I suppose this is what you really want to do here?

Reply:
Thanks, nice find. Yes this should be associated (as it is done in the other parts of the code :) ). Will fix it in a moment ..